modules/auxiliary/scanner/smb/smb_enumshares.rb

```diff
@@ -289,7 +299,16 @@ def run_host(ip)
         smb_login
 
         begin
-          shares = simple.client.net_share_enum_all(ip)
+          # Return all shares if `Share` option has not been set
+          if datastore['Share'].nil?
+            shares = simple.client.net_share_enum_all(ip)
+          else
+            # If the `Share` option has been set attempt to connect to the share to verify it is valid
+            share = "\\\\#{simple.client.default_domain}\\#{datastore['Share']}"
+            simple.client.tree_connect(share)
```

Comment:
This leads to two separate tree connects - one to verify things here, and another when we actually end up connecting to the share; the original comment's goal/intent was to minimize the outgoing network/SMB calls, as a less noise-y client is harder to distinguish/catch

Reply:
I took away the connect from there and just left the one you were referring to. 

Currently the other `tree_connect`'s rescue uses `vprint_error` to check if the connection is valid. Which from my understanding only prints if `verbose` is set to true, would we want to change that to `print_error` or add logic to only use `print_error` if we have a datastore option?

Reply:
As right now it doesn't give the user much to go on in terms of information on why it failed/they received no output:
![image](https://user-images.githubusercontent.com/69522014/195562534-e426ff0d-41de-4f0c-ac29-42d7d0bd89b8.png)

Reply:
Bumping that up to print_error makes sense to me when the user specifies a specified share 👀 

I'm guessing print_verbose made sense when you're enumerating multiple shares - as you would have lots of logging if you weren't able to access any of the shares for example

Reply:
Cool, got that updated. Thanks for all the feedback/help, Appreciate it!